apps/api/src/controllers/post.ts

```diff
@@ -0,0 +1,121 @@
+import Post from "../models/Post";
+import Comment from '../models/Comment';
+//import commentController from '../controllers/comment';
+
+//GET All Posts
+const getPosts = async (req, res) => {
+    try {
+        const posts = await Post.find().populate('category').exec();
+        res.status(200).json(posts);
+    } catch (error) {
+        const {message} = error;
+        res.status(500).send({message}); 
+    }
+}
+
+//GET All Posts bye Category
+const getPostsByCategory = async (req, res) => {
+    const {category} = req.params;
+    console.log(category);
+    try {
+        const posts = await Post.find({category: category});
```

Comment:
For this you will need to add `populate('category') as well to get the category information`